src/main/java/org/kitteh/irc/client/library/implementation/NettyManager.java

```diff
@@ -99,10 +103,16 @@ private ClientConnection(@Nonnull final InternalClient client, @Nonnull ChannelF
                 ClientConnection.this.channelFuture = null;
                 if (future.isSuccess()) {
                     this.buildOurFutureTogether();
+                    this.client.getEventManager().callEvent(new ClientConnectionEstablishedEvent(this.client));
                     this.client.beginMessageSendingImmediate(this.channel::writeAndFlush);
                 } else {
+                    @Nullable final Exception exception = (future.cause() instanceof Exception) ? (Exception) future.cause() : null;
```

Comment:
Why are `Error`s nulled? Shouldn't they be provided in the event too?

Reply:
"An Error is a subclass of Throwable that indicates serious problems that a reasonable application should not try to catch."

I might argue that if there's an error going on, everything's about to crash and burn anyway. Exceptions, like "timed out" "remote host closed connection" or "suddenly Ragnarok", are something you'd want to be tracking while Errors would be logged separately because oh god everything's on fire and my program just ran out of memory.

I'm open to discussing this further, however. That was just my initial reasoning for doing that.

Reply:
following that reasonning, should the client really be allowed to try and reconnect if you don't allow it to handle the exception?

Reply:
I suppose there's no harm in shuttling the errors through, to mirror Netty functionality. Not that there's many scenarios where this would happen. :)